src/cpp/dds/topic/DataWriterImpl.cpp

```diff
@@ -556,6 +556,17 @@ const Publisher* DataWriterImpl::get_publisher() const
 void DataWriterImpl::InnerDataWriterListener::onWriterMatched(
         RTPSWriter* /*writer*/,
         MatchingInfo& info)
+{
+    PublicationMatchedStatus pub_info;
+    pub_info.status = info.status;
+    pub_info.last_subscription_handle = info.remoteEndpointGuid;
+
+    onWriterMatched(data_writer_->writer_, pub_info);
+}
+
+void DataWriterImpl::InnerDataWriterListener::onWriterMatched(
```

Comment:
Apply the same logic than with the DataReaderImpl listener.

Reply:
Solved in #604